mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -0,0 +1,137 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.core.User;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.BlobId;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mailbox.model.search.MailboxQuery;
+import org.apache.james.mailbox.store.StoreMessageManager;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.util.OptionalUtils;
+import org.apache.james.util.streams.Iterators;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class DefaultMailboxBackup implements MailboxBackup {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultMailboxBackup.class);
+
+    protected class MailAccountContent {
+        final MailboxWithAnnotations mailboxWithAnnotations;
+        final Stream<MessageResult> messages;
+
+        MailAccountContent(MailboxWithAnnotations mailboxWithAnnotations, Stream<MessageResult> messages) {
+            this.mailboxWithAnnotations = mailboxWithAnnotations;
+            this.messages = messages;
+        }
+    }
+
+    public DefaultMailboxBackup(MailboxManager mailboxManager, ArchiveService archiveService, BlobStore store) {
+        this.mailboxManager = mailboxManager;
+        this.archiveService = archiveService;
+        this.store = store;
+    }
+
+    @Override
+    public Publisher<BlobId> backupAccount(User user) throws IOException, MailboxException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        List<MailAccountContent> accountContents = getAccountContentForUser(session);
+        List<MailboxWithAnnotations> mailboxes = accountContents.stream().map(accountContent -> accountContent.mailboxWithAnnotations).collect(Guavate.toImmutableList());
```

Comment:
we generally add cr/lf when we chain methods like this (concerns several lines in this file)

Reply:
if you define MailAccountContent.getMailboxWithAnnotations you can:
- make mailboxWithAnnotations field private
- use accountContent::getMailboxWithAnnotations instead of this lambda here